### DISCUSSION TITLE
create new page from website

### DISCUSSION THREAD

**Discussion by dezren39** [srid/emanote#540] [Ideas]
i'd like to be able to make a new page from the website, nothing as complex as #121, more an extension of the 'edit page' functionality that opens github/gitlab editor. i imagine a `+` icon that will make use `<repo>/new/<branch>/<path>` instead of using the `/edit/` link like in the edit icon.

as a separate feature, possibly would be nice to have a quick way to make a new page in _any_ folder, not just the current folder. i imagine some degree of interactivity for that, like how `ctrl+k` works but just for directories-- click 'new item dynamic' icon, type name of folder with autocomplete, then it drops you in github new page url for that path. this would be distinct from #84 because it would work entirely on the static page.

i believe github allows you to pre-populate the file name and even file contents in the 'new' url.
- https://stackoverflow.com/a/27836751/5979634
- `https://github.com/Pithikos/C-Thread-Pool/new/master/tests?filename=newfile.py&value=PUT%20YOUR%20CODE%20HERE`
- `<repo>/new/<branch><path>?filename=<filename>&value=<uri-encoded contents>`

a yaml toggle to pre-populate markdown file with a zettlekasten date would be nice or maybe even a folgezettel id if it can be inferred.

could even populate a header based on the current file header/metadata.

but, to start a 'new file' button would be neat. :-)

**Comment by dezren39** [comment]
one could possibly even allow fancy editing in the static site, then 'submit' the page to github

**Reply by dezren39** [reply]
related? like a miniature version of this, using the static site: https://github.com/srid/ema/discussions/59

**Comment by dezren39** [comment]
separate but related, there is also a `/delete/` url path that could be used to make a 'delete' button on any page. not that i intend to delete that often but it would be a nice little feature to be able to do directly from the website.

**Comment by srid** [comment]
The HTML templates can be edited by the user using [the layer system](https://emanote.srid.ca/guide/layer). We can also expose the necessary template vars from Haskell source (PR welcome) for this.

**Reply by dezren39** [reply]
thank you! i will investigate and may try to make a pr if i can figure anything out.

**Reply by dezren39** [reply]
ok i have this somewhat working but not really clean enough for a pr or anything. i've come up with 2 questions that i am unsure how to go about implementing.
- how can i make index.yaml values accessible in sidebar/sidebar-tree/etc. templates?
  - maybe i can make a new/edit/delete urls in template.sidebar?
- is there a way to reuse stork index with a different url prefix or possibly transforming the route a little? or if needed making an alternate index? i found the stork haskell code that wraps the stork binary but don't really know how it comes together.
